lib/rubocop/cop/style/redundant_to_s.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Checks for redundant uses of `to_s`.
+      #
+      # `to_s` is redundant when called on a string literal, interpolated string, `String.new`,
+      # heredoc, or when chained with itself.
+      #
+      # @safety
+      #   Autocorrect is unsafe because the cop assumes that `to_s` called on an object returns
```

Comment:
Is this something to realistically consider?  `to_s` returning a string seems like a fundamental requirement.

Reply:
I was trying to hedge any bets, but I’m ok with making it safe too. I’d agree that if `to_s` is doing something wonky there’s bigger problems.

Reply:
I think many more cops would need be unsafe if you consider each and every method to be overwritable. It's technically true but I'd just drop this, yeah.

Reply:
Yeah, when considering changes that touch upon the core of the language specification, the number of cops that can truly be deemed "safe" decreases. So this cop should be safe.

Reply:
Sounds good, will do.